OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -307,4 +307,13 @@ public Locomotor(Actor self, LocomotorInfo info)
 			Info = info;
 		}
 	}
-}
+
+	public static class LocomotorExstensions
+	{
+		public static Locomotor GetLocomotor(this World world, MobileInfo mobileInfo)
+		{
+			return world.WorldActor.TraitsImplementing<Locomotor>()
```

Comment:
We would be much better off caching a `public Locomotor Locomotor { get; private set; }` in `Mobile`, which is set once during `Created` instead of querying this over and over